src/server/pfs/server/driver_v2.go

```diff
@@ -144,53 +143,52 @@ func (d *driverV2) getSubFileSet() int64 {
 	return n
 }
 
-// TODO Need commit validation and handling of branch names.
-func (d *driverV2) withUnorderedWriter(ctx context.Context, repo, commit string, cb func(*fileset.UnorderedWriter) error) (retErr error) {
+// withCommitWriter calls cb with an unordered writer. All data written to cb is added to the commit, or an error is returned.
+func (d *driverV2) withCommitWriter(ctx context.Context, repo, commit string, cb func(*fileset.UnorderedWriter) error) (retErr error) {
 	n := d.getSubFileSet()
 	subFileSetStr := fileset.SubFileSetStr(n)
 	subFileSetPath := path.Join(repo, commit, subFileSetStr)
-	fs, err := d.storage.New(ctx, path.Join(tmpPrefix, subFileSetPath), subFileSetStr)
+	id, err := d.withTempUnorderedWriter(ctx, false, cb)
 	if err != nil {
 		return err
 	}
+	tmpPath := path.Join(tmpRepo, id)
+	return d.storage.Copy(ctx, tmpPath, subFileSetPath, defaultTTL)
```

Comment:
Why do the output filesets have a ttl? The output filesets should be permanent.

Reply:
Oops, that would be bad.